### PR TITLE
Coretime Guides Rework (Omninode Considerations)

### PR DESCRIPTION
**Pull request by CrackTheCode016**
_Please note that these only affect the Kusama guides_

**Closes issue #5928**

---

- [x] Add instructions for installing `polkadot-parachain` and `chain-spec-builder`
- [x] Replace using the compiled binary with `polkadot-parachain` as the "omninode"
- [x] Use `chain-spec-builder` as a means for the genesis

### CODE REVIEW

**Comment by CrackTheCode016** [comment]
Will modify this guide once this is merged / resolved: https://github.com/paritytech/polkadot-sdk/pull/4889

File: docs/build/build-guides-install-deps.md
Comment (kianenigma) [inline docs/build/build-guides-install-deps.md]: why not do `cargo install polkadot-parachain-bin`?

File: docs/build/build-guides-template-basic.md
Comment (kianenigma) [inline docs/build/build-guides-template-basic.md:123]: Yesssss, happy to see all of this goes away now 😇

Comment (CrackTheCode016) [reply docs/build/build-guides-template-basic.md:123]: There is still some cleanup to do on my end, I mostly need to back around to the PR to allow for an easier way to generate parachain specs - although, it's just really modifying a JSON after generating it ATM, which isn't terrible, so I may push it with that extra manual step.  WDYT?

Comment (kianenigma) [reply docs/build/build-guides-template-basic.md:123]: agree, I think editing a JSON file is well within the capabilities of anyone. It is not amazing devEx tho, but we can improve that next.

**Comment by CrackTheCode016** [comment]
Besides the trouble I am having with installing `polkadot-parachain` with `cargo install`, the updates to this guide should be sound and working. 

There is still a lot of improvement to be made.  For example, the steps should be broken down a little bit more into seperate pages for reserving, template-related items, and deploying.  I don't think the final place for this guide - so this is a good stopping point.

**Comment by CrackTheCode016** [comment]
The issues above have been fixed. Someone else should run through this guide and ensure it works - I tested most parts, but not a full deployment (yet). 

There is still a good amount of work to improve this guide, but we can save that for docs.polkadot.com ;) This is a good stopping point for now that uses the latest versions and direction.

File: docs/build/build-guides-coretime-start.md
Comment (profradha) [inline docs/build/build-guides-coretime-start.md]:
```suggestion
The Polkadot SDK is comprised of **three** important repositories:
```

File: docs/build/build-guides-coretime-start.md
Comment (profradha) [inline docs/build/build-guides-coretime-start.md]:
```suggestion
You might see the term "task" referenced often in place of "parachain". In most cases, it refers to a
```

File: docs/build/build-guides-coretime-start.md
Comment (profradha) [inline docs/build/build-guides-coretime-start.md]:
```suggestion
process, provided that it adheres to the Polkadot protocol interface.
```

File: docs/build/build-guides-coretime-start.md
Comment (profradha) [inline docs/build/build-guides-coretime-start.md]:
```suggestion
Be sure you also install the `polkadot-parachain` and `chain-spec-builder` binaries, as they needed to start and run your chain!
```

File: docs/build/build-guides-template-basic.md
Comment (profradha) [inline docs/build/build-guides-template-basic.md]:
```suggestion
This guide is considered a moving document and currently uses the **Rococo** testnet. This guide is also applicable to the parachains on the Kusama
relay chain, as coretime is also enabled there. Polkadot will
```

File: docs/build/build-guides-template-basic.md
Comment (profradha) [inline docs/build/build-guides-template-basic.md]:
```suggestion
**This instructions on this guide are applicable for the Polkadot SDK repository with tag `polkadot-v1.15.1`**
```

File: docs/build/build-guides-template-basic.md
Comment (profradha) [inline docs/build/build-guides-template-basic.md]:
```suggestion
which is mirrored in the templates folder within Polkadot SDK repository.
```

File: docs/build/build-guides-template-basic.md
Comment (profradha) [inline docs/build/build-guides-template-basic.md]:
```suggestion
The chain specification is a JSON file that describes Polkadot SDK-based networks. It usually contains
```

File: docs/build/build-guides-template-basic.md
Comment (profradha) [inline docs/build/build-guides-template-basic.md]:
```suggestion
the genesis runtime (in hex) under `genesis.runtimeGenesis.code` and also contains
genesis values/state for the pallets included in your runtime.
```

File: docs/build/build-guides-template-basic.md
Comment (profradha) [inline docs/build/build-guides-template-basic.md]:
```suggestion
Feel free to use the patch provided here, which you can look to tweak to your liking.
```

File: docs/build/build-guides-template-basic.md
Comment (profradha) [inline docs/build/build-guides-template-basic.md]:
```suggestion
Generate the genesis following the instructions below:
```

File: docs/build/build-guides-template-basic.md
Comment (profradha) [inline docs/build/build-guides-template-basic.md]: This paragraph could improve with a bit of elaboration on the sync process. What do `blocks-pruning` and `state-pruning` do? even a small context on these flags will improve the reader's experience.

The tone is conversational and could be a bit more instructional. It's better to avoid terms like ASAP in technical documentation.

"could take some time depending on your download speed" -> "the sync process can benefit from higher network bandwidth as it is network resource intensive" or something along those lines

**Comment by CrackTheCode016** [comment]
@DrW3RK I think this guide is actually already out of date due to the following factors: 

- Rococo is being deprecated
- chain-spec-builder should be parachain enabled (to some extent, still waiting for it to be included in a release)
- Overall writing can be improved with what we've learned over the past months

Perhaps we work to merge this, then work on these updates? WDYT?

**Review by DrW3RK — APPROVED** [review]
Agreed

File: docs/build/build-guides-template-basic.md
Comment (profradha) [inline docs/build/build-guides-template-basic.md]:
```suggestion
`state-pruning` is only used for the first initial sync for the database.
```

**Pull request merged**
